fx-src/FirefoxStorage.js

```diff
@@ -72,7 +72,12 @@ const statements = {
   "listRecords": `
     SELECT record
       FROM collection_data
-        WHERE collection_name = :collection_name;`
+        WHERE collection_name = :collection_name;`,
+
+  "importData": `
+    REPLACE INTO collection_data (collection_name, record_id, record)
```

Comment:
Hi Richard,
Thank you for this thorough review!

Regarding this point, we can explain the inconsistency between the doc up there and this piece of code.

Indeed, the high-level (and two-phase) checking of existing records versus imported records is done in the `Kinto/collection.js` class, so that it does not have to be duplicated in every adapter (IndexedDB, localstorage and SQLite). Therefore the code is the adapter is rather naive and brutal because its input comes prepared and filtered already.

In this repo we run the collection tests against indexeddb and localstorage. Thanks to your feedback we added the same tests cases in mozilla-central (see https://github.com/Natim/gecko-dev/commit/dc9c6a80e51738c7d4246073fade6879068e6570, @Natim is submitting a patch in BZ https://bugzilla.mozilla.org/show_bug.cgi?id=1227877)